contracts/GNTDeposit.sol

```diff
@@ -137,7 +137,7 @@ contract GNTDeposit is ReceivingContract, Ownable {
         emit Lock(msg.sender);
     }
 
-    function onTokenReceived(address _from, uint256 _amount, bytes /* _data */) public onlyToken {
+    function onTokenReceived(address _from, uint256 _amount, bytes memory /* _data */) public onlyToken {
```

Comment:
The the `memory` exactly means? How does it work when receiving an external call?

Reply:
This `memory` here is just for clarity, it's not like you can put anything else here. For `public` methods it has to be `memory` while for `external` it's `calldata`. And that means that for an external here the calldata will be copied to memory and cost some gas.
At least that's my understanding.

Reply:
Why this function is not `external` in the first place?
If the `data` param is not used, can you make it `calldata`?